sjsonnet/src/sjsonnet/StdRegex.scala

```diff
@@ -4,66 +4,58 @@ import sjsonnet.Expr.Member.Visibility
 import sjsonnet.Val.Obj
 
 object StdRegex {
+  def regexPartialMatch(pos: Position, pattern: String, str: String): Val = {
+    val compiledPattern = Platform.getPatternFromCache(pattern)
+    val matcher = compiledPattern.matcher(str)
+    var returnStr: Val = null
+    val groupCount = matcher.groupCount()
+    val captures = Array.newBuilder[Val]
+    captures.sizeHint(groupCount)
+
+    while (matcher.find()) {
```

Comment:
On closer look, I'm not actually sure that we want a `while()` loop here: I think we only intend to find the _first_ match, not all? Let me try adding a test case to see if I can demonstrate a bug from the loop behavior.

Reply:
I think our unit test cases are wrong:

```
{
    res: std.regexPartialMatch("a(b)c", "cabc")
}
```

in jrsonnet master built with the experimental regex support enabled yields:

```
{
   "res": {
      "captures": [
         "b"
      ],
      "namedCaptures": { },
      "string": "abc"
   }
}
```

Reply:
We might also want to pull in all of the test cases from `google/jsonnet` PR, too, or at least make sure that we have exact parity for their edge cases.